arch/sim/src/sim/up_uart.c

```diff
@@ -31,12 +31,16 @@
 
 #include "up_internal.h"
 
-#if defined(USE_DEVCONSOLE) || CONFIG_SIM_UART_NUMBER > 0
+#if defined(USE_DEVCONSOLE) || defined(CONFIG_SIM_UART)
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
 
-#define BUFSIZE 256
+#ifdef CONFIG_SIM_UART_BUFFER_SIZE
```

Comment:
Just as an alternative we can replace `BUFSIZE` with `CONFIG_SIM_UART_BUFFER_SIZE` and have here
```
#ifndef CONFIG_SIM_UART_BUFFER_SIZE
#  define CONFIG_SIM_UART_BUFFER_SIZE 256
#endif
```

Reply:
But seems that `CONFIG_SIM_UART_BUFFER_SIZE` now is always defined

Reply:
I wouldn't like to do it like this. I believe that `CONFIG_` macros shall only be defined in Kconfigs, otherwise it will be a mess.

As I see, `BUFSIZE` is also used when `USE_DEVCONSOLE` is selected, even if no UARTs are configured (and thus `CONFIG_SIM_UART_BUFFER_SIZE` is not set).

Reply:
Can we still go with
```
#ifndef CONFIG_SIM_UART_BUFFER_SIZE
# define CONFIG_SIM_UART_BUFFER_SIZE 256
#endif
```
??

Reply:
@pkarashchenko Sorry, I didn't understand.  
What will be the benefit of this?

When needed, `CONFIG_SIM_UART_BUFFER_SIZE` is automatically defined by Kconfig, why we need this check?

Reply:
The benefit is to eliminate `BUFSIZE` and switch to one define instead of two

Reply:
let's change to:
```
#define BUFSIZE CONFIG_SIM_UART_BUFFER_SIZE
```
or use CONFIG_SIM_UART_BUFFER_SIZE directly?

Reply:
@fjpanag do you want to apply the suggested change?

Reply:
@fjpanag please change to 
#ifndef CONFIG_SIM_UART_BUFFER_SIZE
# define CONFIG_SIM_UART_BUFFER_SIZE 256
#endif
as @pkarashchenko suggested

there is no need to keep the BUFSIZE

Reply:
I'm pushing this change today.  
My only concern was that defining a `CONFIG_*` macro in a place other than a Kconfig may be confusing, with no real benefit.

Nevertheless, I will proceed as suggested.